src/Data/Bin/Properties.agda

```diff
@@ -1,143 +1,904 @@
+                                                     (sym (toℕ*homo a c))
+                                        ⟩
+  fromℕ (toℕ (a * b) +ₙ toℕ (a * c))   ≡⟨ cong fromℕ $
+                                            sym (toℕ+homo (a * b) (a * c))
+                                           ⟩
+  fromℕ (toℕ (a * b + a * c))           ≡⟨ fromℕ∘toℕ (a * b + a * c) ⟩
+  a * b + a * c                           ∎
+  where
+  k = toℕ a;   m = toℕ b;   n = toℕ c
+
+
+*-distribʳ-+ :  _DistributesOverʳ_ _*_ _+_
+*-distribʳ-+ a b c = begin
+  (b + c) * a      ≡⟨ *-comm (b + c) a ⟩
+  a * (b + c)      ≡⟨ *-distribˡ-+ a b c ⟩
+  a * b + a * c    ≡⟨ cong₂ _+_ (*-comm a b) (*-comm a c) ⟩
+  b * a + c * a    ∎
+
+*-distrib-+ :  _DistributesOver_ _*_ _+_
+*-distrib-+ =  (*-distribˡ-+ , *-distribʳ-+)
+
+2*-distrib : (x y : Bin) → 2* (x + y) ≡ 2* x + 2* y
+2*-distrib x y = begin
+  2* (x + y)            ≡⟨ 2*≗2#* (x + y) ⟩
+  2# * (x + y)          ≡⟨ *-distribˡ-+ 2# x y ⟩
+  (2# * x) + (2# * y)   ≡⟨ cong₂ _+_ eq eq' ⟩
+  2* x + 2* y           ∎
+  where
+  eq = sym (2*≗2#* x);   eq' = sym (2*≗2#* y)
+
+
+*-identityˡ :  (1# *_) ≗ id
+*-identityˡ x = begin
+  1# * x                   ≡⟨ sym (fromℕ∘toℕ (1# * x)) ⟩
+  fromℕ (toℕ (1# * x))   ≡⟨ cong fromℕ (toℕ*homo 1# x) ⟩
+  fromℕ (1 *ₙ n)          ≡⟨ cong fromℕ (ℕp.*-identityˡ n) ⟩
+  fromℕ n                 ≡⟨ fromℕ∘toℕ x ⟩
+  x                        ∎
+  where n = toℕ x
+
+*-identityʳ :  (_* 1#) ≗ id
+*-identityʳ x =  trans (*-comm x 1#) (*-identityˡ x)
+
+2#*x≡x+x :  (x : Bin) → 2# * x ≡ x + x
+2#*x≡x+x x = begin
+  2# * x            ≡⟨⟩
+  (1# + 1#) * x     ≡⟨ *-distribʳ-+ x 1# 1# ⟩
+  1# * x + 1# * x   ≡⟨ cong₂ _+_ (*-identityˡ x) (*-identityˡ x) ⟩
+  x + x             ∎
+
+2x≡x+x :  (x : Bin) → 2* x ≡ x + x
+2x≡x+x x =  trans (2*≗2#* x) (2#*x≡x+x x)
+
+[1+x]* :  (x y : Bin) → (1# + x) * y ≡ y + x * y
+[1+x]* x y = begin
+  (1# + x) * y     ≡⟨ *-distribʳ-+ y 1# x ⟩
+  1# * y + x * y   ≡⟨ cong (_+ (x * y)) (*-identityˡ y) ⟩
+  y + x * y        ∎
+
+*[1+x] :  (x y : Bin) → y * (1# + x) ≡ y + y * x
+*[1+x] x y = begin
+  y * (1# + x)     ≡⟨ *-distribˡ-+ y 1# x ⟩
+  y * 1# + y * x   ≡⟨ cong (_+ (y * x)) (*-identityʳ y) ⟩
+  y + y * x        ∎
+
+suc* :  (x y : Bin) → (suc x) * y ≡ y + x * y
+suc* x y = begin
+  (suc x) * y     ≡⟨ cong (_* y) (suc≗1+ x) ⟩
+  (1# + x) * y    ≡⟨ [1+x]* x y ⟩
+  y + x * y       ∎
+
+*suc :  (x y : Bin) → x * (suc y) ≡ x + x * y
+*suc x y = begin
+  x * (suc y)    ≡⟨ cong (x *_) (suc≗1+ y) ⟩
+  x * (1# + y)   ≡⟨ *[1+x] y x ⟩
+  x + x * y      ∎
+
+------------------------------------------------------------------------------
+-- Multiplicative structure instances
+------------------------------------------------------------------------------
+
+*-isMagma :  IsMagma _≡_ _*_
+*-isMagma =  record{ isEquivalence =  isEquivalence
+                   ; ∙-cong        =  cong₂ _*_
+                   }
+
+*-magma :  Magma 0ℓ 0ℓ
+*-magma =  record{ Carrier = Bin
```

Comment:
You can omit all of these lines except the `isMagma` line.

Reply:
Done a similar omitting in many places.